couchbase/src/api/collections.rs

```diff
@@ -0,0 +1,324 @@
+use crate::io::request::*;
+use crate::io::Core;
+use crate::CouchbaseError::{CollectionExists, CollectionNotFound, ScopeExists, ScopeNotFound};
+use crate::{
+    CouchbaseError, CouchbaseResult, CreateCollectionOptions, CreateScopeOptions,
+    DropCollectionOptions, DropScopeOptions, ErrorContext, GenericManagementResult,
+    GetAllScopesOptions,
+};
+use futures::channel::oneshot;
+use serde_derive::Deserialize;
+use serde_json::Value;
+use std::sync::Arc;
+use std::time::Duration;
+
+#[derive(Debug)]
+pub struct ScopeSpec {
+    name: String,
+    collections: Vec<CollectionSpec>,
+}
+
+impl ScopeSpec {
+    pub fn new(name: String, collections: Vec<CollectionSpec>) -> Self {
```

Comment:
if these are public, for convenience it could make sense to not accept string but rather S: Into<String>, then someone can either provide a string or a &'static str like ::new("foo")